### PR TITLE
🗑️ Rewrite org deletion test and fix deletion issues

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/orgs/models.py [hunk 1]
```diff
@@ -1332,9 +1332,8 @@ def delete(self):
         self.urns.all().delete()
 
         # delete our fields
-        for contactfield in self.fields.all():
-            contactfield.release(user)
-            contactfield.delete()
+        for field in self.fields.all():
+            field.delete()
```
Comment (rowanseymour) [inline temba/orgs/models.py:1336]: for fields, callin `release` isn't needed and just blows up if field is still being used in a dynamic group

File: temba/orgs/models.py [hunk 2]
```diff
@@ -1274,8 +1275,9 @@ def delete(self):
         Does an actual delete of this org
         """
 
-        assert not self.is_active and self.released_on, "can't delete an org which hasn't been released"
-        assert not self.deleted_on, "can't delete an org twice"
+        assert not self.is_active and self.released_on, "can't delete org which hasn't been released"
+        assert self.released_on < timezone.now() - timedelta(days=7), "can't delete org which was released recently"
```
Comment (rowanseymour) [inline temba/orgs/models.py:1279]: make it that little bit harder to accidentally delete an org...

File: temba/orgs/models.py [hunk 3]
```diff
@@ -1349,15 +1350,13 @@ def delete(self):
 
             channel.delete()
 
-        for g in self.globals.all():
-            g.release(user)
+        for glob in self.globals.all():
+            glob.delete()
```
Comment (rowanseymour) [inline temba/orgs/models.py:1354]: actually delete globals

**Comment by susanm74** [comment]
👍

**Pull request merged**